modules/custom/falcon_mail/src/Plugin/Mail/FalconMailSystem.php

```diff
@@ -0,0 +1,138 @@
+<?php
+
+namespace Drupal\falcon_mail\Plugin\Mail;
+
+use Drupal;
+use Exception;
+use Drupal\Component\Utility\Html;
+use Drupal\Core\Mail\Plugin\Mail\PhpMail;
+
+/**
+ * Modify the drupal mail system to use theme template and replace tokens when sending emails.
+ *
+ * @Mail(
+ *   id = "falcon_mail_system",
+ *   label = @Translation("Falcon Mailer"),
+ *   description = @Translation("Provides formatter that can wrap emails into HTML templates and replace tokens.")
+ * )
+ */
+class FalconMailSystem extends PhpMail {
+
+  /**
+   * Concatenate and wrap the e-mail body for either HTML emails.
+   *
+   * @param array $message
+   *   A message array.
+   *
+   * @return array
+   *   The formatted $message.
+   */
+  public function format(array $message) {
+    $message['subject'] = !empty($message['subject']) ? $message['subject'] :
+      (isset($message['params']['subject']) ? $message['params']['subject'] : '');
+    $message['body'] = !empty($message['body']) ? $message['body'] :
+      (isset($message['params']['body']) ? $message['params']['body'] : []);
+    $message['from'] = !empty($message['from']) ? $message['from'] :
+      (isset($message['params']['from']) ? $message['params']['from'] : Drupal::config('system.site')->get('mail'));
+
+    // Merge headers.
+    if (!empty($message['params']['headers'])) {
+      $message['headers'] = array_merge($message['headers'], $message['params']['headers']);
+    }
+
+    if (is_array($message['body'])) {
+      $message['body'] = implode("\n\n", $message['body']);
+    }
+
+    // Wrap body in theme template.
+    if (!empty($message['params']['theme_template'])) {
+      $message['body'] = str_replace('#$#BODY#$#', $message['body'], $message['params']['theme_template']);
```

Comment:
Could you please make `#$#BODY#$#` a constant? Something like `self::BODY_TOKEN`

Reply:
Done.